src/plugins/workspace/server/plugin.ts

```diff
@@ -67,12 +77,51 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
     });
   }
 
+  private setupPermission(core: CoreSetup, { applicationConfig }: AppPluginSetupDependencies) {
+    this.permissionControl = new SavedObjectsPermissionControl(this.logger);
+
+    core.http.registerOnPostAuth(async (request, response, toolkit) => {
+      let groups: string[];
+      let users: string[];
+      let configGroups: string[];
+      let configUsers: string[];
+
+      // There may be calls to saved objects client before user get authenticated, need to add a try catch here as `getPrincipalsFromRequest` will throw error when user is not authenticated.
+      try {
+        ({ groups = [], users = [] } = this.permissionControl!.getPrincipalsFromRequest(request));
+      } catch (e) {
+        return toolkit.next();
+      }
+
+      if (!!applicationConfig) {
+        [configGroups, configUsers] = await getApplicationOSDAdminConfig(
+          { applicationConfig },
+          request
+        );
+      } else {
+        [configGroups, configUsers] = await getOSDAdminConfig(this.globalConfig$);
+      }
+      updateDashboardAdminStateForRequest(request, groups, users, configGroups, configUsers);
+      return toolkit.next();
+    });
+
+    this.workspaceSavedObjectsClientWrapper = new WorkspaceSavedObjectsClientWrapper(
+      this.permissionControl
+    );
+
+    core.savedObjects.addClientWrapper(
+      0,
```

Comment:
we should use `PRIORITY_FOR_PERMISSION_CONTROL_WRAPPER` right?

Reply:
Yes, I will fix it.